SMTInterpol/src/de/uni_freiburg/informatik/ultimate/smtinterpol/proof/MinimalProofChecker.java

```diff
@@ -1012,14 +1036,19 @@ void stackPush(final ProofLiteral[] pushClause, final Term keyTerm) {
 		}
 		case ":" + ProofRules.DT_CONS: {
 			if (!theory.getLogic().isDatatype()) {
-				throw new AssertionError();
+				reportError("Proof requires data type theory");
+				return getTrueClause(theory);
 			}
 			assert annots.length == 1;
 			final Term[] params = (Term[]) annots[0].getValue();
 			assert params.length == 1;
 			final ApplicationTerm isConsTerm = (ApplicationTerm) params[0];
 			if (!isConsTerm.getFunction().getName().equals(SMTLIBConstants.IS)) {
-				throw new AssertionError();
+				final StringBuilder sb = new StringBuilder();
+				sb.append("Side-condition does not hold in ");
+				ProofRules.printProof(sb, axiom);
+				reportError(sb.toString());
+				return getTrueClause(theory);
```

Comment:
Why not use reportViolatedSideCondition(axiom) ?

Reply:
fixed